code/modules/xenoarcheaology/effect_master.dm

```diff
@@ -0,0 +1,379 @@
+
+/*
+ * Here there be dragons.
+ */
+
+/atom/var/datum/artifact_master/artifact_master
+
+/atom/Initialize()
+	. = ..()
+
+	if(ispath(artifact_master))
```

Comment:
I would very strongly recommend against this. Many atoms don't call the entire Initialize() chain, some deliberately, and putting it here means _every other non-datum non-constant in the game_ is going to be doing this check during init, for the sake of what looks like is primarily for adminbuse?

This is the exact use-case that components and extensions were intended for, and Polaris definitely seems to have components - it might be worth looking into them rather than tighly coupling it to atoms.